### ISSUE TITLE
"git repository does not have an annex uuid" - but it does?!

### ISSUE THREAD

**Issue by julia-pfarr** [datalad/datalad#7387]
### What is the problem?

Hi,
I created datasets locally and want to push them to GIN hosted by our university (gin-prod.hrz.uni-marburg.de). Creating a sibling wasn't an issue, but when I try running ```datalad siblings configure -s gin --as-common-datasrc gin-storage``` it gives me the following error:
```
CommandError: 'git -c diff.ignoreSubmodules=none annex initremote gin-storage type=git location=https://gin-prod.hrz.uni-marburg.de/pfarr/myrepo autoenable=true -c annex.dotfiles=true' failed with exitcode 1 under /Users/julia-katharinapfarr/Desktop/Test_2
initremote gin-storage 

failed
  https://gin-prod.hrz.uni-marburg.de/pfarr/myrepo/config download failed: Not Found
git-annex: git repository does not have an annex uuid
initremote: 1 failed
```

But my config file shows this:
```
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	ignorecase = true
	precomposeunicode = true
[annex]
	uuid = c284cda8-03a5-4c0c-8f51-1d1e60619b25
	version = 10
[filter "annex"]
	smudge = git-annex smudge -- %f
	clean = git-annex smudge --clean -- %f
	process = git-annex filter-process
[submodule "sub-01"]
	active = true
	url = ./sub-01
[remote "gin"]
	annex-ignore = false
	url = https://gin-prod.hrz.uni-marburg.de/pfarr/myrepo
	fetch = +refs/heads/*:refs/remotes/gin/*
	annexurl = git@gin-prod.hrz.uni-marburg.de:/pfarr/myrepo.git
	pushurl = git@gin-prod.hrz.uni-marburg.de:/pfarr/myrepo.git
	annex-uuid = 0dfd7e19-4ad9-4ca1-8978-0dc343add680
``` 
which makes me think that there is an annex uuid, no?

So now I see in my GIN repository my files on the main branch, but when I click on my subdataset with the actual imaging file, I get an 404. 

Does anyone have an idea what I'm doing wrong or where to look for further clues? Thanks!!!

### DataLad information

```
# WTF
## configuration <SENSITIVE, report disabled by configuration>
## credentials 
  - keyring: 
    - active_backends: 
      - macOS Keyring
      - PlaintextKeyring with no encyption v.1.0 at /Users/julia-katharinapfarr/.local/share/python_keyring/keyring_pass.cfg
    - config_file: /Users/julia-katharinapfarr/.config/python_keyring/keyringrc.cfg
    - data_root: /Users/julia-katharinapfarr/.local/share/python_keyring
## datalad 
  - version: 0.18.3
## dataset 
  - branches: 
    - git-annex@4a9a8b5
    - main@ae9f376
  - id: 893f1485-7a63-4a0b-8cfa-7f414179a03b
  - path: /Users/julia-katharinapfarr/Desktop/Test_2
  - repo: AnnexRepo
## dependencies 
  - annexremote: 1.6.0
  - boto: 2.49.0
  - cmd:7z: 17.05
  - cmd:annex: 10.20230407
  - cmd:bundled-git: UNKNOWN
  - cmd:git: 2.37.3
  - cmd:ssh: 9.0p1
  - cmd:system-git: 2.37.3
  - cmd:system-ssh: 9.0p1
  - humanize: 4.6.0
  - iso8601: 1.1.0
  - keyring: 23.13.1
  - keyrings.alt: 4.2.0
  - msgpack: 1.0.5
  - platformdirs: 3.1.1
  - requests: 2.28.2
## environment 
  - LANG: en_US.UTF-8
  - PATH: /usr/local/fsl/bin:/Users/julia-katharinapfarr/miniconda3/bin:/Users/julia-katharinapfarr/miniconda3/condabin:/Library/Frameworks/Python.framework/Versions/3.10/bin:/usr/local/fsl/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/opt/X11/bin
## extensions 
## git-annex 
  - build flags: 
    - Assistant
    - Webapp
    - Pairing
    - FsEvents
    - TorrentParser
    - MagicMime
    - Benchmark
    - Feeds
    - Testsuite
    - S3
    - WebDAV
  - dependency versions: 
    - aws-0.24
    - bloomfilter-2.0.1.0
    - cryptonite-0.30
    - DAV-1.3.4
    - feed-1.3.2.1
    - ghc-9.4.4
    - http-client-0.7.13.1
    - persistent-sqlite-2.13.1.1
    - torrent-10000.1.3
    - uuid-1.3.15
    - yesod-1.6.2.1
  - key/value backends: 
    - SHA256E
    - SHA256
    - SHA512E
    - SHA512
    - SHA224E
    - SHA224
    - SHA384E
    - SHA384
    - SHA3_256E
    - SHA3_256
    - SHA3_512E
    - SHA3_512
    - SHA3_224E
    - SHA3_224
    - SHA3_384E
    - SHA3_384
    - SKEIN256E
    - SKEIN256
    - SKEIN512E
    - SKEIN512
    - BLAKE2B256E
    - BLAKE2B256
    - BLAKE2B512E
    - BLAKE2B512
    - BLAKE2B160E
    - BLAKE2B160
    - BLAKE2B224E
    - BLAKE2B224
    - BLAKE2B384E
    - BLAKE2B384
    - BLAKE2BP512E
    - BLAKE2BP512
    - BLAKE2S256E
    - BLAKE2S256
    - BLAKE2S160E
    - BLAKE2S160
    - BLAKE2S224E
    - BLAKE2S224
    - BLAKE2SP256E
    - BLAKE2SP256
    - BLAKE2SP224E
    - BLAKE2SP224
    - SHA1E
    - SHA1
    - MD5E
    - MD5
    - WORM
    - URL
    - X*
  - local repository version: 10
  - operating system: darwin x86_64
  - remote types: 
    - git
    - gcrypt
    - p2p
    - S3
    - bup
    - directory
    - rsync
    - web
    - bittorrent
    - webdav
    - adb
    - tahoe
    - glacier
    - ddar
    - git-lfs
    - httpalso
    - borg
    - hook
    - external
  - supported repository versions: 
    - 8
    - 9
    - 10
  - upgrade supported from repository versions: 
    - 0
    - 1
    - 2
    - 3
    - 4
    - 5
    - 6
    - 7
    - 8
    - 9
    - 10
  - version: 10.20230407
## location 
  - path: /Users/julia-katharinapfarr/Desktop/Test_2
  - type: dataset
## python 
  - implementation: CPython
  - version: 3.11.3
## system 
  - distribution: darwin/22.1.0 13.0/x86_64
  - encoding: 
    - default: utf-8
    - filesystem: utf-8
    - locale.prefered: UTF-8
  - filesystem: 
    - CWD: 
      - path: /Users/julia-katharinapfarr/Desktop/Test_2
    - HOME: 
      - path: /Users/julia-katharinapfarr
    - TMP: 
      - path: /var/folders/th/f4r354c54vs4hw5w6lf0z87c0000gn/T
  - max_path_length: 298
  - name: Darwin
  - release: 22.1.0
  - type: posix
  - version: Darwin Kernel Version 22.1.0: Sun Oct  9 20:14:54 PDT 2022; root:xnu-8792.41.9~2/RELEASE_X86_64
```

### Additional context

I tried running ```datalad install``` from the remote repo into a new empty folder and I do get an output when running ```datalad subdatasets```, so something worked. However, if I then run ```datalad get .``` it tells me ```GIN: Repository does not exist or you do not have access``` and 
```
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.']]
``` 

### Have you had any success using DataLad before?

Yes, following workshops and working with the example repos from the workshops.

**Comment by celprov** [comment]
@julia-pfarr how did you solve this issue?
I'm running into the same problem

**Comment by julia-pfarr** [comment]
> @julia-pfarr how did you solve this issue? I'm running into the same problem

I actually don't remember, to be honest, I'm sorry!...I think my issue was something about that I didn't have the correct access rights for the local files

**Comment by yarikoptic** [comment]
rright -- I think I saw similar ones with that recent git change where it forbids access to repos owned by other people and demands adding them to `safe.directory`. 

There was also a bug on git-annex side fixed since then:
- https://git-annex.branchable.com/bugs/safe.bareRepository_breaks_git-annex_uuid/

fixed in [10.20230626-58-ga05bc6a314 AKA 10.20230802~99](https://git.kitenet.net/index.cgi/git-annex.git/commit/?id=a05bc6a31459dedd5d199fbc864af2e9d7d1f6d1) so check if you have up to date git-annex
